src/components/manager/access/GrantForm.tsx

```diff
@@ -0,0 +1,514 @@
+    const groupOptions = useMemo(() => groups.map((g: StoredGroup) => ({
+        value: g.id,
+        label: <><strong>{g.name}</strong> (ID: {g.id})</>,
+    })), [groups]);
+
+    const onChangeGroup = useCallback((v: number) => {
+        setGroup(v);
+        handleSubjectChange(onChange, subjectType, iss, sub, client, v);
+    }, onChangeDeps);
+
+    return <Space direction="vertical" style={{ width: "100%", minHeight: 32 }}>
+        <Radio.Group value={subjectType} onChange={onChangeSubjectType} options={subjectTypeOptions} />
+        {(subjectType === "iss-sub" || subjectType === "iss-client") && (
+            <Space style={{ width: "100%" }} styles={{ item: { flex: 1 } }}>
+                <Input placeholder="Issuer URI" value={iss} onChange={onChangeIssuer} />
+                {subjectType === "iss-sub" ? (
+                    <Input placeholder="Subject ID" value={sub} onChange={onChangeSubject} />
+                ) : (
+                    <Input placeholder="Client ID" value={client} onChange={onChangeClient} />
+                )}
+            </Space>
+        )}
+        {subjectType === "group" && (
+            <Select
+                showSearch={true}
+                placeholder="Select a group"
+                value={group}
+                options={groupOptions}
+                onChange={onChangeGroup}
+            />
+        )}
+    </Space>;
+};
+
+
+const buildResource = (
+    rt: "everything" | "project-plus",
+    p: Project | null,
+    d: Dataset | null,
+    dt: string,
+): GrantResource => {
+    if (rt === "everything" || p === null) {
+        return RESOURCE_EVERYTHING;
+    }
+
+    const res: GrantResource = { "project": p.identifier };
+
+    if (d) res["dataset"] = d?.identifier;
+    if (dt) res["data_type"] = dt;
+
+    return res;
+};
+
+type ResourceInputProps = {
+    value?: GrantResource,
+    onChange?: (value: GrantResource) => void;
+};
+
+const ResourceInput = ({ value, onChange }: ResourceInputProps) => {
+    // TODO: consolidate when useProjects() is typed
+    const projects: Project[] = useProjects().items;
+    const projectsByID: Record<string, Project> = useProjects().itemsByID;
+    const datasetsByID: Record<string, Dataset> = useProjects().datasetsByID;
+    const dataTypes: BentoServiceDataType[] = useDataTypes().items;
+
+    const [resourceType, setResourceType] = useState<"everything" | "project-plus">("everything");
+    const [selectedProject, setSelectedProject] = useState<Project | null>(null);
+    const [selectedDataset, setSelectedDataset] = useState<Dataset | null>(null);
+    const [selectedDataType, setSelectedDataType] = useState<string>("");
+
+    useEffect(() => {
+        if (!value) return;
+        if ("everything" in value) {
+            setResourceType("everything");
+        } else {
+            // TODO: how to handle missing projects?
+            setSelectedProject(projectsByID[value.project]);
+            // TODO: how to handle missing datasets?
```

Comment:
Not sure I get the TODOs. If "everything" is not in `value`, then project should be defined according to `Resource` type.

Reply:
this means "what if the project has been deleted"

Reply:
(right now, it doesn't matter since we cannot edit grants)